### ISSUE TITLE
from 1.6 to 1.7 tests throwing : updating the path … would create a conflict at …

### ISSUE THREAD

**Issue by Guema** [meteor/meteor#10168]
So, im trying to pass from 1.6 to 1.7 but im encountering some errors in tests im unable to solve or even understand.
more details [here](https://stackoverflow.com/questions/51827085/migrating-to-meteor-1-7-0-4-mongodb-updating-the-path-would-create-a-co)

To make a story short summary :

I have got my tests (and only tests) throwing, on some specific collection fields, and only since i passed 1.7 (do not reproduce in 1.6)

the error messages are the followings :

```
Error: Updating the path 'metadata' would create a conflict at 'metadata'
    at Function.MongoError.create (C:\Users\fquesselaire\AppData\Local\.meteor\packages\npm-mongo\3.0.11\npm\node_modules\mongodb-core\lib\error.js:45:10)
    at toError (C:\Users\fquesselaire\AppData\Local\.meteor\packages\npm-mongo\3.0.11\npm\node_modules\mongodb\lib\utils.js:149:22)
    at C:\Users\fquesselaire\AppData\Local\.meteor\packages\npm-mongo\3.0.11\npm\node_modules\mongodb\lib\collection.js:1029:39
    at C:\Users\fquesselaire\AppData\Local\.meteor\packages\npm-mongo\3.0.11\npm\node_modules\mongodb-core\lib\connection\pool.js:544:18
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9)
```

```
Error: Updating the path 'relationChangeRate' would create a conflict at 'relationChangeRate'
    at Function.MongoError.create (C:\Users\fquesselaire\AppData\Local\.meteor\packages\npm-mongo\3.0.11\npm\node_modules\mongodb-core\lib\error.js:45:10)
    at toError (C:\Users\fquesselaire\AppData\Local\.meteor\packages\npm-mongo\3.0.11\npm\node_modules\mongodb\lib\utils.js:149:22)
    at C:\Users\fquesselaire\AppData\Local\.meteor\packages\npm-mongo\3.0.11\npm\node_modules\mongodb\lib\collection.js:1029:39
    at C:\Users\fquesselaire\AppData\Local\.meteor\packages\npm-mongo\3.0.11\npm\node_modules\mongodb-core\lib\connection\pool.js:544:18
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9)
```
They occur multiple times in tests

i was using :
- External MongoDB 3.4.last
- Meteor 1.6.1.3

i tried :
- External MongoDB 3.4.last
- Meteor 1.7.0.4

i use now :
- External MongoDB 3.6.6
- Meteor 1.7.0.4

(tried to update MongoDB, because i thank that version mismatch could cause this, but their was no effect)
(My development server works on Meteor's included MongoDB version, but has the same problems on tests) 

Notes :
Im using aldeed:collection2-core/collection2, dburles:collection-helpers, simple-schema, and a monkey-patched version of practicalmeteor:mocha-core to make it compatible with TeamCity (tried without, no changes in local) as server-signifiant packages.
See [StackOverFlow post for more details](https://stackoverflow.com/questions/51827085/migrating-to-meteor-1-7-0-4-mongodb-updating-the-path-would-create-a-co)

**Comment by MaxGuitet** [comment]
We have noticed similar issues in our app using ``, _aldeed:collection2_, and _simpl-schema_. When validating, the _simpl-schema` package [adds a field `$setOnInsert`](https://github.com/aldeed/simple-schema-js/blob/ee2ce4bc0b56762e7892a6c73d2c2caaeb00e83f/package/lib/SimpleSchema.js#L910). What I noticed is that on the `before.update` hook, if you delete the `$setOnInsert` hook, you don't have the error. So I think the conflicts comes from the field being set in both `$set` and `$setOnInsert`. It wasn't an issue in previous mongo versions, but seems to have become one lately.
